spec/facter/query_parser_spec.rb

```diff
@@ -79,6 +79,25 @@
         contain_exactly(an_instance_of(Facter::SearchedFact).and(having_attributes(fact_class: path_class)))
     end
 
+    context 'when user query contains metacharacters (like dots)' do
+      it 'finds the structured fact and not the alias' do
+        query_list = ['ldom.domainname.impl']
+        ldom_class = 'Facter::Solaris::Ldom'
+
+        loaded_fact_ldom = double(Facter::LoadedFact, name: 'ldom', klass: ldom_class, type: :core, file: nil)
+        ldom_fact_ldom_alias = double(Facter::LoadedFact, name: 'ldom_domainrole_impl', klass: ldom_class,
```

Comment:
I noticing the test still passed when I reverted the `query_parser.rb` fix, I think because we're querying for `domainname` on line 84, but stubbing `domainrole` here?

```suggestion
        ldom_fact_ldom_alias = double(Facter::LoadedFact, name: 'ldom_domainname_impl', klass: ldom_class,
```

Reply:
Shoot, keep on making small mistakes. I fixed it so it's `domainrole` and the test no longer passes when the changes in `query_parser.rb` are reverted but will pass with the changes